src/foam/mlang/mlang.js

```diff
@@ -866,6 +866,24 @@ foam.CLASS({
         this.upperCase_ = nu && foam.core.Enum.isInstance(nu);
       }
     },
+    {
+      name: 'arg2',
+      preSet: function(old, nu) {
+        if ( ! ( this.Constant.isInstance(nu) &&
+                 Array.isArray(nu.value) ) ) {
+          return nu;
+        }
+
+        // Convert enum value arg2 values to strings for uppercase comparison.
+        var array = nu.value;
+        for ( var i = 0; i < array.length; i++ ) {
+          if ( foam.core.AbstractEnum.isInstance(array[i]) )
```

Comment:
Why do we do this?

Reply:
Lines 899-907: Constant value on right will be treated as array of strings and used as keys for a value set. This detects when an enum value, rather than its string name, is used in the array on the right.

We need actual strings for the toUpperCase() case. I guess the alternative would be to use toString().toUpperCase() below. 

@kgrgreer  WDTY?